server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -275,6 +295,26 @@ public HealthIndicatorResult calculate(boolean explain) {
         MIGRATE_TO_TIERS_ACTION_GUIDE
     );
 
+    public static final Map<String, UserAction.Definition> ACTION_MIGRATE_TIERS_AWAY_FROM_INCLUDE_DATA_LOOKUP = DataTier.ALL_DATA_TIERS
+        .stream()
+        .collect(
+            Collectors.toUnmodifiableMap(
+                tier -> tier,
+                tier -> new UserAction.Definition(
+                    "migrate_data_tiers_include_data_" + tier,
+                    "Elasticsearch isn't allowed to allocate some shards from these indices to any nodes in the ["
+                        + tier
+                        + "] data tiers because the indices are configured with allocation filter rules that are incompatible with the "
```

Comment:
```suggestion
                        + "] data tier because the indices are configured with allocation filter rules that are incompatible with the "
```